src/lib/users/ssh_authorized_keyring.rb

```diff
@@ -0,0 +1,213 @@
+#  GNU General Public License for more details.
+#
+#  You should have received a copy of the GNU General Public License
+#  along with this program; if not, contact SUSE LLC.
+#
+#  To contact SUSE about this file by physical or electronic mail,
+#  you may find current contact information at www.suse.com
+
+require "yast"
+require "users/ssh_authorized_keys_file"
+
+module Yast
+  module Users
+    # Read, write and store SSH authorized keys.
+    #
+    # This class manages authorized keys for each home directory in the
+    # system.
+    class SSHAuthorizedKeyring
+      include Yast::Logger
+
+      # @return [Hash<String,Array<SSHAuthorizedKey>>] Authorized keys indexed by home directory
+      attr_reader :keys
+      private :keys
+
+      # The home directory does not exist.
+      class HomeDoesNotExist < StandardError; end
+
+      # Constructor
+      def initialize
+        @keys = {}
+      end
+
+      # Add/register a keys
+      #
+      # This method does not make any change to the system. For that,
+      # see #write_keys.
+      #
+      # @param home [String] Home directory where the key will be stored
+      # @return [Array<SSHAuthorizedKey>] Registered authorized keys
+      def add_keys(home, new_keys)
+        keys[home] = new_keys
+      end
+
+      # Returns the keys for a given home directory
+      #
+      # @return [Array<SSHAuthorizedKey>] List of authorized keys
+      def [](home)
+        keys[home] || []
+      end
+
+      # Determines if the keyring is empty
+      #
+      # @return [Boolean] +true+ if it's empty; +false+ otherwise
+      def empty?
+        keys.empty?
+      end
+
+      # Read keys from a given home directory and add them to the keyring
+      #
+      # @param path [String] User's home directory
+      # @return [Boolean] +true+ if some key was found
+      def read_keys(home)
+        path = authorized_keys_path(home)
+        return false unless FileUtils::Exists(path)
+        authorized_keys = Yast::Users::SSHAuthorizedKeysFile.new(path).keys
+        keys[home] = authorized_keys unless authorized_keys.empty?
+        log.info "Read #{authorized_keys.size} keys from #{path}"
+        !authorized_keys.empty?
+      end
+
+      # Write user keys to the given file
+      #
+      # If SSH_DIR does not exist in the given directory, it will be
+      # created inheriting owner/group and setting permissions to SSH_DIR_PERM.
+      #
+      # @param path [String] User's home directory
+      # @return [Boolean] +true+ if keys were written; +false+ otherwise
+      def write_keys(home)
+        return false if keys[home].nil?
+        if !FileUtils::Exists(home)
+          log.error("Home directory '#{home}' does not exist!")
+          raise HomeDoesNotExist
+        end
+        owner, group = perms_from(home)
+        if create_ssh_dir(home, owner, group, SSH_DIR_PERMS)
+          write_file(home, owner, group, AUTHORIZED_KEYS_PERMS)
```

Comment:
Yes, I follow certain *functional mindset* for this kind of helpers method. I'll remove them.